File: Java/14.Longest_Common_Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String res = "";
        if (strs.length == 0)
            return res;
        for (int i = 0; i < strs[0].length(); i++) {
            String letter = strs[0].substring(i, i + 1);
            for (String str : strs) {
                if (i < str.length() && letter.equals(str.substring(i, i + 1)))
                    ;
                else
                    return res;
            }
            res += letter;
        }
        return res;
    }
}

File: Python3/14.Longest_Common_Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        total = ""
        if len(strs) == 0:
            return total
        for i in range(len(strs[0])):
            char = strs[0][i]
            for str in strs:
                if i < len(str) and char == str[i]:
                    pass
                else:
                    return total
            total += char
        return total
